client/src/stores/toolStore.ts

```diff
@@ -177,6 +180,20 @@ export const useToolStore = defineStore("toolStore", () => {
         }
     }
 
+    async function fetchPanelViews() {
+        if (defaultPanelView.value && Object.keys(panelViews.value).length > 0) {
+            return;
+        }
+        try {
+            const { data } = await axios.get(`${getAppRoot()}api/tool_panels`);
+            const { default_panel_view, views } = data;
+            defaultPanelView.value = default_panel_view;
+            panelViews.value = views;
+        } catch (e) {
+            rethrowSimple(e);
+        }
+    }
```

Comment:
I think there's an edge case here, if `fetchPanelViews` is called from two places at once. A simple boolean flag can fix this.